m4/check_pkg_cuda.m4

```diff
@@ -21,40 +21,52 @@ AC_DEFUN([CHECK_PKG_CUDA], [
         [],
         [test "${with_cuda}" = "no"],
         [check_pkg_found=no],
-        [AS_IF([test -d ${with_cuda}/lib64], [check_pkg_libdir="lib64"], [check_pkg_libdir="lib"])
+        [AS_IF([test -d $(realpath ${with_cuda})/lib64], [check_pkg_libdir="lib64"], [check_pkg_libdir="lib"])
+         CUDA_LDFLAGS="-L$(realpath ${with_cuda})/${check_pkg_libdir}"
+         CUDA_LIBS="-lcudart_static -lrt -ldl"
+         CUDA_CPPFLAGS="-fPIC -I$(realpath $(realpath ${with_cuda})/include)"
```

Comment:
Why is the realpath needed here and not everywhere else?

-fPIC should never be in CPPFLAGS.  It's a CFLAG.

Reply:
`/usr/local/cuda` is a symlink to a specific `/usr/local/cuda-12.5`. `/usr/local/cuda/lib` and/or `/usr/local/cuda/lib64` are symlinks to `targets/x86_64-linux/include`. This makes those paths explicit when reading config.log/build failure output.

Reply:
But we don't do that anywhere else and have never felt the need to do it before.  I don't know that there's a huge win, but I guess whatever.